src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -492,6 +492,7 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
         }
 
         internal static ImmutableArray<IArgument> MakeArgumentsInEvaluationOrder(
+            CSharpOperationFactory operationFactory,
```

Comment:
>CSharpOperationFactory operationFactory, [](start = 12, length = 40)

I do not think this change is necessary, we should cache entire result of this method instead of doing it one argument in a time.